sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java

```diff
@@ -278,7 +278,10 @@ private static void logOutcome(StringBuilder stringBuilder, EvaluationOutcome ou
             Collection<Protos.TaskInfo> allTasks) {
         Map<String, ResourceSet> resourceSets = getNewResourceSets(podInstanceRequirement);
 
-        Optional<TLSEvaluationStage.Builder> tlsBuilder = getTLSEvaluationStageBuilderFromEnvironment(schedulerFlags);
+        Optional<TLSEvaluationStage.Builder> tlsBuilder = Optional.empty();
+        if (!TaskUtils.getTasksWithTLS(podInstanceRequirement).isEmpty()) {
```

Comment:
One could also modify `getTLSEvaluationStageBuilderFromEnvironment` to accept the podInstanceRequirements as well and perform this conditional check. Was there a specific reason for not doing this? I only ask, because this would have required fewer changes in the two places where the function is called?